### PR TITLE
check for state of the controller node before cleanup

### PR DESCRIPTION
**Pull request by saikishor**
Fixes https://github.com/ros-controls/ros2_control/issues/1136

If a controller tries to do the cleanup from an unconfigured state, it prints the following error in the terminal 

```
Unable to start transition 2 from current state unconfigured: Transition is not registered., at ./src/rcl_lifecycle.c:355

### CODE REVIEW

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -664,7 +664,20 @@ controller_interface::return_type ControllerManager::unload_controller(
...
+  {
+    RCLCPP_DEBUG(
+      get_logger(), "Controller '%s' is cleaned-up before unloading!", controller_name.c_str());
+    // TODO(destogl): remove reference interface if chainable; i.e., add a separate method for
+    // cleaning-up controllers?
+    const auto new_state = controller.c->get_node()->cleanup();
+    if (new_state.id() != lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+    {
+      RCLCPP_WARN(
+        get_logger(), "Unable to clean-up the controller '%s' before unloading!",
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp:677]:
```suggestion
        get_logger(), "Failed to clean-up the controller '%s' before unloading!",
```

Comment (saikishor) [reply controller_manager/src/controller_manager.cpp]: Done

**Review by destogl — APPROVED** [review]
It looks good. We should also follow up with the #1236

**Comment by saikishor** [comment]
> It looks good. We should also follow up with the #1236 

@destogl I just left a review on the other PR as well. Mostly, it looks good to me

**Comment by firesurfer** [comment]
@saikishor I tried to test the fix this morning but ran into the issue that the whole controller_manager crashes on my `iron` setup if I build it from source. 

But I took a quick look at the changes you made and they look reasonable to me.

**Review by bmagyar — APPROVED** [review]

**Pull request merged**